File: PythonClient/ai.py
# -*- coding: utf-8 -*-

# python imports
import random

# chillin imports
from chillin_client import RealtimeAI

# project imports
from ks.models import World
from ks.commands import Move, Turn, Fire
import random


class AI(RealtimeAI):

    def __init__(self, world):
        super(AI, self).__init__(world)


    def initialize(self):
        pass


    def decide(self):
        # print('decide')
        for medic in self.world.medics[self.my_side]:
            # here is stupid medic`s code!!!
            random_args = [
                (medic.id, random.uniform(0, medic.max_move_distance)),
                (medic.id, random.choice([True, False]), random.uniform(0, medic.max_turn_angle)),
                (medic.id, random.choice([True, False]), random.uniform(0, medic.max_turn_angle))
            ]
            commands = [self.move, self.turn, self.fire]
            random_command = random.randint(0, 2)
            commands[random_command](*random_args[random_command])


    def move(self, medic_id, distance):
        self.send_command(Move(medic_id, distance))


    def turn(self, medic_id, clockwise, angle):
        self.send_command(Turn(medic_id, clockwise, angle))


    def fire(self, medic_id, clockwise, angle):
        self.send_command(Fire(medic_id, clockwise, angle))

File: CppClient/Game/ai.cpp
#include "ai.h"

#include <ctime>
#include <vector>
#include <iostream>

using namespace std;
using namespace koala::chillin::client;
using namespace ks::models;
using namespace ks::commands;


AI::AI(World *world): RealtimeAI<World*>(world)
{
    srand(time(0));
}

void AI::initialize()
{
}

void AI::decide()
{
    for (int i = 0; i < this->world->ref_medics()[this->mySide].size(); i++)
    {
        Medic medic = this->world->ref_medics()[this->mySide][i];
        int rand = getRandInt(0, 2);

        if (rand == 0)
        {
            this->move(
                medic.id(),
                getRandFloat(0, medic.max_move_distance())
            );
        }
        else if (rand == 1)
        {
            this->turn(
                medic.id(),
                (bool) getRandInt(0, 1),
                getRandFloat(0, medic.max_turn_angle())
            );
        }
        else
        {
            this->fire(
                medic.id(),
                (bool) getRandInt(0, 1),
                getRandFloat(0, medic.max_fire_angle())
            );
        }
    }
}

int AI::getRandInt(int start, int end)
{
    return (rand() % (end - start + 1)) + start;
}

float AI::getRandFloat(float start, float end)
{
    float n = ((float) rand()) / ((float) RAND_MAX);
    return n * (end - start) + start;
}


void AI::move(int medicId, float distance)
{
    Move cmd;
    cmd.id(medicId);
    cmd.distance(distance);
    this->sendCommand(&cmd);
}

void AI::turn(int medicId, bool clockwise, float angle)
{
    Turn cmd;
    cmd.id(medicId);
    cmd.clockwise(clockwise);
    cmd.angle(angle);
    this->sendCommand(&cmd);
}

void AI::fire(int medicId, bool clockwise, float angle)
{
    Fire cmd;
    cmd.id(medicId);
    cmd.clockwise(clockwise);
    cmd.angle(angle);
    this->sendCommand(&cmd);
}
